### DISCUSSION TITLE
Linking to cURL

### DISCUSSION THREAD

**Discussion by zorael** [community/community#27189] [Actions]
<p>I’m trying to set up a workflow to test my project with <code>runs-on: ubuntu-latest</code>, and it fails to link due to missing libraries.</p>
<p><a href="https://github.com/zorael/kameloso/blob/ee2a50b2bd2742a195290a65c81e44a28b32ac91/.github/workflows/d.yml" rel="noopener nofollow ugc">Workflow .yml</a><br>
<a href="https://github.com/zorael/kameloso/runs/1560669227?check_suite_focus=true" rel="noopener nofollow ugc">Example log</a></p>
<pre><code class="lang-auto">Linking...
/usr/bin/ld: cannot find -lcurl
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
dmd failed with exit code 1.
</code></pre>
<p>I tried adding a seemingly suitable cURL action (<code>wei/curl</code>), and it installed the command-line <code>curl</code> tool, but no library to link to.</p>
<p>Is it possible to get it to <code>apt install libcurl-dev</code>? I really could not find any guide mentioning missing libraries.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can use <code>sudo</code> without a password to install additional packages like on any Ubuntu system, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#administrative-privileges-of-github-hosted-runners">Administrative privileges of GitHub-hosted runners</a>.</p>
